### DISCUSSION TITLE
Recommended syntax for slicing arrays

### DISCUSSION THREAD

**Discussion by josephtoles** [hylang/hy#2186] [Q&A] [answered]
I use Hy for data science. This requires a lot of cutting/slicing. In particular, I frequently have to cut a list or an array from an index until the end. In Python, the way to slice an array from an index to the end is to leave the second index argument of a slice empty as in `arra[5:]`. In the Hy stable release, I used `cut` with a single argument.

This no longer works in Hy's alpha. According to the Hy alpha's changelog, "`(cut x n)` gets the first `n` elements instead of everything after the first `n`."

What is the recommended method for getting everything after the first `n` elements?

**Comment by allison-casey** [comment, marked as answer]
`cut` was changed to have the same syntax as `slice` which is what python's `:` sugar compiles down to.  So `arr[5:]` actually compiles down to `arr[slice(5, None)]`. Now that cut is equivalent all the following are identical.
```clojure
(setv x [1 2 3 4 5 6 7])
(get x (slice 5 None))
;; => [6 7]
(cut x 5 None)
;; => [6 7]
```
```python
x[5:]
# => [6, 7]
x[slice(5, None)]
# => [6, 7]
```

If you need more complex slicing mechanics (or more "Pythonic"), and you're willing to use the master branch Hy version, you can check out Hyrule's `#:` and `ncut` [macros](https://hyrule.readthedocs.io/en/master/#hyrule.collections.#:). Specifically `ncut` for you since it allows the sugared n-dimensional slicing that `numpy` and `pandas` use frequently
```clojure
(get x #: 5:)
;; => [6 7]
(ncut x 5:)
;; => [6 7]
```

**Comment by josephtoles** [comment]
Thank you so much. I didn't realize I could add `None` as the second argument to `slice`. I also didn't know what `ncut` did since I only read the stable version's documentation.
